lib/screens/play/card/card_actions/card_action_panel.dart

```diff
@@ -0,0 +1,117 @@
+import 'package:flutter/material.dart';
+
+import 'package:treasure_of_the_high_seas/model/card/action/card_action.dart'
+    as Model;
+import 'package:treasure_of_the_high_seas/model/card/action/card_action_details.dart';
+import 'package:treasure_of_the_high_seas/model/game_result.dart';
+import 'package:treasure_of_the_high_seas/model/game_state.dart';
+import 'package:treasure_of_the_high_seas/model/resource.dart';
+
+import './card_action_text.dart';
+import 'card_action_line.dart';
+
+class CardActionPanel extends StatelessWidget {
+  final Model.CardAction action;
+  final GameState state;
+  final bool readOnly;
+
+  CardActionPanel(this.action, this.state, { this.readOnly = false });
+
+  @override
+  Widget build(BuildContext context) {
+    final actionDetails = action?.actionDetails;
+    final enabled = action?.isEnabled(state) ?? false;
+
+    final shape = RoundedRectangleBorder(
+      side: BorderSide(color: Colors.black, width: 1),
+      borderRadius: BorderRadius.circular(10),
+    );
+
+    return Padding(
+        padding: EdgeInsets.symmetric(vertical: 4),
+        child: readOnly
+            ? Card(
+                shape: shape,
+                color: _getButtonColor(actionDetails),
+                child: _getActionComponents(actionDetails))
+            : RaisedButton(
+                shape: shape,
+                onPressed: enabled
+                    ? () {
+                        action?.performAction(state);
+                      }
+                    : null,
+                color: _getButtonColor(actionDetails),
+                child: _getActionComponents(actionDetails)));
+  }
+
+  Column _getActionComponents(CardActionDetails actionDetails) {
+    if (actionDetails == null) {
+      return null;
+    }
+
+    final effects = <CardActionLine>[];
+
+    if (actionDetails.cost.isNotEmpty) {
+      final description = '${actionDetails.cost.map((res) => res.getText())}';
+      effects
+          .add(CardActionLine(description, Icons.arrow_downward, Colors.red));
+    }
+
+    if (actionDetails.reward.isNotEmpty) {
+      final description = '${actionDetails.reward.map((res) => res.getText())}';
+      effects
+          .add(CardActionLine(description, Icons.arrow_upward, Colors.green));
+    }
+
+    if (actionDetails.cardsToScry != null) {
+      effects.add(CardActionLine("Scry ${actionDetails.cardsToScry}",
+          Icons.remove_red_eye, Colors.blue));
+    }
+
+    if (actionDetails.replacement != null) {
+      effects.add(CardActionLine("${actionDetails.replacement.name}",
+          Icons.description, Colors.purpleAccent));
+    }
+
+    return Column(
+      children: [
+        SizedBox(height: 5),
```

Comment:
Using `SizedBox` here effectively as padding. Found conflicting advice around online, but this seems perhaps preferable to wrapping all our components in `Padding` elements, which causes extra nesting.